### ISSUE TITLE
Finetune CogVideoX-5B (not 1.5) T2V takes 3 hours for 1 step using 4090

### ISSUE THREAD

**Issue by MaulanaYusufIkhsanRobbani** [THUDM/CogVideo#771]
Hello. I try to finetune CogVideoX-5B (not 1.5) T2V and it takes 3 hours to do 1 step using RTX 4090. Is it normal thing?
`You set `add_prefix_space`. The tokenizer needs to be converted from the slow tokenizers
Downloading shards: 100%|███████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 1433.71it/s]
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████| 2/2 [00:09<00:00,  4.75s/it]
Fetching 2 files: 100%|██████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 180.33it/s]
05/04/2025 10:37:58 - INFO - trainer - Initialized Trainer
05/04/2025 10:37:58 - INFO - trainer - Accelerator state:
Distributed environment: NO
Num processes: 1
Process index: 0
Local process index: 0
Device: cuda

Mixed precision type: bf16

05/04/2025 10:37:58 - INFO - trainer - Initializing models
05/04/2025 10:37:58 - INFO - trainer - Initializing dataset and dataloader
05/04/2025 10:38:03 - INFO - trainer - Precomputing latent for video and prompt embedding ...
05/04/2025 10:38:03 - INFO - trainer - Precomputing latent for video and prompt embedding ... Done
05/04/2025 10:38:05 - INFO - trainer - Initializing trainable parameters
05/04/2025 10:38:06 - INFO - trainer - Initializing optimizer and lr scheduler
05/04/2025 10:38:09 - INFO - trainer - Initializing trackers
05/04/2025 10:38:09 - INFO - trainer - Starting training
05/04/2025 10:38:09 - INFO - trainer - Memory before training start: {
    "memory_allocated": 21.256,
    "memory_reserved": 21.262,
    "max_memory_allocated": 21.256,
    "max_memory_reserved": 21.262
}
05/04/2025 10:38:09 - INFO - trainer - Training configuration: {
    "trainable parameters": 132120576,
    "total samples": 5,
    "train epochs": 10,
    "train steps": 50,
    "batches per device": 1,
    "total batches observed per epoch": 5,
    "train batch size total count": 1,
    "gradient accumulation steps": 1
}
Training steps:   2%|▍                  | 1/50 [3:19:29<162:50:51, 11964.33s/it, grad_norm=0.00366, loss=0.063, lr=2e-7]`
Here is my variables:
`
    "--model_path", "THUDM/CogVideoX-5B", #THUDM/CogVideoX1.5-5B
    "--model_name", "cogvideox-t2v", #cogvideox1.5-t2v
    "--model_type", "t2v",
    "--training_type", "lora",
    "--output_dir", "/home/ikhsanrobbani/output_dir",
    "--report_to", "tensorboard",
    "--data_root", "/home/ikhsanrobbani/datasets/ProductVisualisationDatasetSmall2",
    "--caption_column", "prompt.txt",
    "--video_column", "videos.txt",
    "--train_resolution", "49x480x720", #81x
    "--train_epochs", "10",
    "--seed", "42",
    "--batch_size", "1",
    "--gradient_accumulation_steps", "4", #1
    "--mixed_precision", "bf16",
    "--num_workers", "8", #8
    "--pin_memory", "True",
    "--nccl_timeout", "1800",
    "--checkpointing_steps", "10",
    "--checkpointing_limit", "2",
    "--do_validation", "false",
    "--validation_dir", "/home/ikhsanrobbani/datasets/ProductVisualisationDatasetSmall2/val",
    "--validation_steps", "20",
    "--validation_prompts", "prompts.txt",
    "--gen_fps", "8"`
And this is my accelerate config file
`compute_environment: LOCAL_MACHINE

gpu_ids: "0,1,2,3,4,5,6,7"
num_processes: 1 #8  # should be the same as the number of GPUs

debug: false
deepspeed_config:
  deepspeed_config_file: /mnt/d/TA_ucup/CogVideo/finetune/configs/zero2.yaml  # e.g. configs/zero2.yaml, need use absolute path
  zero3_init_flag: false
distributed_type: DEEPSPEED
downcast_bf16: 'no'
enable_cpu_affinity: false
machine_rank: 0
main_training_function: main
num_machines: 1
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false`

**Comment by OleehyO** [comment]
This seems quite odd. Does the official example dataset also exhibit this behavior? Additionally, for LoRA training, we recommend using the DDP strategy (rather than Zero).

**Comment by MaulanaYusufIkhsanRobbani** [comment]
I use the Disney-VideoGeneration-Dataset also takes a long time.
The output:
Training steps:   2%|▍                      | 1/54 [43:50<38:43:26, 2630.32s/it, grad_norm=0.00171, loss=0.203, lr=1e-5]
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████| 2/2 [00:15<00:00,  7.93s/it]
Fetching 2 files: 100%|████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 66052.03it/s]
05/13/2025 20:39:25 - INFO - trainer - Initialized Trainer
05/13/2025 20:39:25 - INFO - trainer - Accelerator state:
Distributed environment: NO
Num processes: 1
Process index: 0
Local process index: 0
Device: cuda

Mixed precision type: bf16

05/13/2025 20:39:25 - INFO - trainer - Initializing models
05/13/2025 20:39:25 - INFO - trainer - Initializing dataset and dataloader
05/13/2025 20:39:30 - INFO - trainer - Precomputing latent for video and prompt embedding ...
05/13/2025 20:39:31 - INFO - trainer - Precomputing latent for video and prompt embedding ... Done
05/13/2025 20:39:33 - INFO - trainer - Initializing trainable parameters
05/13/2025 20:39:33 - INFO - trainer - Initializing optimizer and lr scheduler
05/13/2025 20:39:39 - INFO - trainer - Initializing trackers
05/13/2025 20:39:39 - INFO - trainer - Starting training
05/13/2025 20:39:39 - INFO - trainer - Memory before training start: {
    "memory_allocated": 20.782,
    "memory_reserved": 20.805,
    "max_memory_allocated": 20.782,
    "max_memory_reserved": 20.805
}
05/13/2025 20:39:39 - INFO - trainer - Training configuration: {
    "trainable parameters": 8257536,
    "total samples": 69,
    "train epochs": 3,
    "train steps": 54,
    "batches per device": 1,
    "total batches observed per epoch": 69,
    "train batch size total count": 4,
    "gradient accumulation steps": 4
}
Training steps:   4%|█▍                                     | 2/54 [2:06:02<39:29:05, 2733.58s/it, loss=0.0662, lr=2e-5]

**Comment by MaulanaYusufIkhsanRobbani** [comment]
btw I use train_ddp_t2v.sh script
